### DISCUSSION TITLE
OpenAPI integration with Jolie self-contained REST services

### DISCUSSION THREAD

**Discussion by mwallnoefer** [jolie/jolie#479] [Ideas (Features, Fixes, etc.)]
As I have explored Jolie capabilities for creating self-contained REST services over the HTTP osc calls I have wondered how to generate an OpenAPI spec. According to the docs this seems impossible, since the _jolie2openapi_ tool interacts exclusively with _Jester_, the more powerful alternative for REST in Jolie which requires an external rest call descriptor (_rest_template.json_).

So my question is as follows: does there exist some plan to implement an OpenAPI-compatible export feature also for self-contained REST services? I guess that it could make sense, since the solution seems easier.

**Comment by klag** [comment]
Hi Matthias, you are right. So far it is not possible to create an OpenAPI spec starting from the osc parameters in the http protocol. At the same time, actually, jester is not necessary because the tool just analyzes the template.json file. I think it is sufficient to extract the template from the osc parameters and then reusing the same tool.

If you are interested in developing this part, I can help you,
